src/Ast/AST.php

```diff
@@ -426,6 +426,36 @@ public function toCode(): string
         };
     }
 
+    /**
+     * Create a try/catch/finally statement.
+     *
+     * @param array $tryCode The code to use in the try block.
+     *
+     * @return AST
+     */
+    public static function try(...$tryCode): AST
+    {
+        return new class($tryCode) extends AST {
+            public function __construct($tryCode)
+            {
+                $this->tryCode = AST::block(...$tryCode);
+                $this->finallyCode = null;
+            }
+            public function finally(...$finallyCode): AST
+            {
+                return $this->clone(fn($clone) => $clone->finallyCode = AST::block(...$finallyCode));
+            }
+            public function toCode(): string
+            {
+                return
+                    "try {\n" .
+                    static::toPhp($this->tryCode) . "\n" .
+                    '}' .
+                    (is_null($this->finallyCode) ? '' : "finally {\n" . static::toPhp($this->finallyCode) . "\n}");
```

Comment:
What does try without catch or finally do, out of interest?

Reply:
Nothing, it's invalid ;) So it'll always be required to add a `finally` or `catch` part. But so far I haven't needed to generate any `catch`es, so that code isn't yet implemented.